### ISSUE TITLE
Increase max rendering distance 

### ISSUE THREAD

**Issue by OctoNezd** [colobot/colobot#857]
200% is not enough

**Comment by krzys-h** [comment]
:+1:, GPUs nowadays are able to handle much more than when the original game was released

**Comment by RaptorParkowsky** [comment]
200% currently actually allow us to see almost half of the standard level heightmap if there's no DeepView distance limit set in the level. I don't think this is really necessary to increase this 200% to something like 500%, because then you'll see the level borders which are ugly.

I think rewriting rendering distance setting would be more effective. Let's just make that 100% setting actually would allow us to make whole standard 161x161 heightmap level visible, but DeepView setting in the level's scene should be still decisive.

**Comment by krzys-h** [comment]
@RaptorParkowsky There are options to change the level size, and I don't think that level size influence this graphical setting.

The border is ugly even in the current version. Is there a way we could add fog just at the borders to hide them when you are far away, and show them only if you come really close? @tomaszkax86 ?

**Comment by RaptorParkowsky** [comment]
About borders I actually thought about something like enhancing inaccessible areas in existing level heightmaps. Maybe as random enhancing or maybe repeating like that:

![schowek01](https://cloud.githubusercontent.com/assets/4019345/20215936/1ddb5dd8-a818-11e6-91d4-0845caec04d2.png)

The landscape behind that selected area couldn't be seen.

**Comment by tomaszkax86** [comment]
We could do something like in GTA:SA, where outside of actual map there's infinite water. For some levels we could add infinite ground surface. However, for obvious reasons we should add invisible walls at the border, so players can't actually leave the map.

**Comment by RaptorParkowsky** [comment]
@tomaszkax86 Actually there should be combined methods for infinite water and ground as not every heightmap represents island. Depend of heightmap there's water or mount at the edge of the level.

Invisible walls at the border are there already, but they need only some kind of explanation. Someone once proposed just to display some kind of message like "This is the end of the mission area. Please don't pass through."

**Comment by tomangelo2** [comment]
Maybe something like it was in another old game, when player goes outside the level, there was at first warning about going outside mission area, and if you ignored it and goes further, then you heard something about "you escaped, mission failed", and mission was already failed. 
All without invisible walls, player would just got information from NASA about "we have some issues with tracking you, please go back to mission area", and then "we lost you, mission failed" or something like that, for robots we could just have information "robot escaped, you lost control on him", and it'll just disappear.

**Comment by tomangelo2** [comment]
An idea for these "far lands". Somewhere in the code is relief generator. So these far lands would be generated with it, making quite realistic terrain, without strange looking repeating, or full flat lands.

**Comment by RaptorParkowsky** [comment]
@tomangelo2 Yep, but there will be serious problems with seamless looking of the terrain. Currently relief generator creates just basic noise without any natural procedures to create natural looking landscapes.

Possibly only by manually prepared relief extension maps we could achieve such natural looking inaccessible areas.
